### PR TITLE
Remove dependency on derivative

### PR DESCRIPTION
**Pull request by djc**
A custom Debug impl is pretty straightforward for this.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -76,15 +76,23 @@ pub mod tokio_adapter;
...
-#[derivative(Debug)]
 pub struct StackDriverExporter {
-  #[derivative(Debug = "ignore")]
   tx: futures::channel::mpsc::Sender<Vec<Arc<SpanData>>>,
   pending_count: Arc<AtomicUsize>,
   maximum_shutdown_duration: Duration,
 }
 
+impl fmt::Debug for StackDriverExporter {
+  fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
```
Comment (ErichDonGubler) [inline src/lib.rs]: Super pedantic 🚲: this `fmt` trait method arg is called `f` [upstream](https://doc.rust-lang.org/std/fmt/trait.Debug.html) -- IMO there's usually not a reason to _not_ align with that. Thoughts?

File: src/lib.rs [hunk 2]
```diff
@@ -76,15 +76,23 @@ pub mod tokio_adapter;
...
+    fmt
+      .debug_struct("StackDriverExporter")
+      .field("tx", &"(elided)")
+      .field("pending_count", &self.pending_count)
+      .field("maximum_shutdown_duration", &self.maximum_shutdown_duration)
```
Comment (ErichDonGubler) [inline src/lib.rs]: One concern I see with trading a `derive`d `Debug` implementation with this hand-rolled one that specifically accesses fields via `self` is that a `derive` will never accidentally miss a newly added field, which with this code is a distinct possibility. There's a solution, though: destructuring `self`.

Comment (ErichDonGubler) [reply src/lib.rs]: My suggested edit here:

```suggestion
    #[allow(clippy::unneeded_field_pattern)]
    let Self {
      maximum_shutdown_duration,
      pending_count,
      tx: _,
    } = self;
    fmt
      .debug_struct("StackDriverExporter")
      .field("tx", &"(elided)")
      .field("pending_count", pending_count)
      .field("maximum_shutdown_duration", maximum_shutdown_duration)
```

**Review by ErichDonGubler — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (djc) [reply src/lib.rs]: I see no reason not to align with, so I have -- I somehow thought it was usually called `fmt`, but I just found out that most `Debug` impls in Quinn (one larger codebase I'm familiar with) calls most of them `f`. Seemed like good fodder for [a Twitter poll](https://twitter.com/djco/status/1326508180357189633), too.

**Comment by djc** [comment]
Also, thanks for the review -- if you have time to review the other PRs I made to this project, would be much appreciated!

**Comment by Xaeroxe** [comment]
Thanks @ErichDonGubler you raised the one concern I had here. I'm going to merge this.

**Pull request merged**